### ISSUE TITLE
Event Queries searches all indexes in the cluster

### ISSUE THREAD

**Issue by cw-pratik** [Graylog2/graylog2-server#6910]
When you create an Event with a search query, The query runs on all indexes present in the cluster (Even the indexes that are not a part of graylog index sets).

## Expected Behavior

Create an event in graylog. Add a filter search query and select a stream. The search query should run only in the indexes that are defined in the graylog index sets. It should not run on indexes that are not a part of graylog. This will not only give a wrong result if the index mapping is similar it will also consume a lot of time.

## Current Behavior

I have only one index set in graylog which has indexes with a prefix "graylog_*" 

I have created some events for which we need notifications. The search happens in all indexes.

I noticed this as there were some fields with different datatypes present in the cluster and it filled the log files on elasticsearch.

## Possible Solution

Restrict the search to only graylog defined indexes (The same thing works properly when it comes to direct searching)

* Graylog Version: 3.1.0 
* Elasticsearch Version: 6.5.4
* MongoDB Version: 3.6.10
* Operating System: ubuntu 16.04
* Browser version:

**Comment by mpfz0r** [comment]
Thanks for you report.
I can't see how this would happen and can't reproduce it.
Do you have additional information? Maybe the search request that gets run against elastic?

**Comment by cw-pratik** [comment]
Have a look at the event that is configured. 

![graylogEvent](https://user-images.githubusercontent.com/15033612/70238322-0fbc9200-178f-11ea-8f47-8aa55e1cca42.JPG)

The query is 

`Level:ERROR AND Referrer:*\/advantage\/* AND (NOT messageOrEvent:"Script error" OR NOT messageOrEvent:"Script error.")
`

This are the indices configured 

![GraylogIndices](https://user-images.githubusercontent.com/15033612/70238470-527e6a00-178f-11ea-969f-319f126e69c3.JPG).

There is only one index that we use that is the default index.

These are the logs that we get in elasticsearch in large volumes (10 every 3-4 seconds)

```
[2019-12-03T11:13:09,968][DEBUG][o.e.a.s.TransportSearchAction] [graylog3] [%{[@metadata][beat]}-2018.06.03][0], node[TyQNq21DRBOiOVkEbWHvLw], [P], s[STARTED], a[id=bPTJ_CebQEeSKZiKBEVKWg]: Failed to execute [SearchRequest{searchType=QUERY_THEN_FETCH, indices=[], indicesOptions=IndicesOptions[ignore_unavailable=false, allow_no_indices=false, expand_wildcards_open=true, expand_wildcards_closed=false, allow_aliases_to_multiple_indices=true, forbid_closed_indices=true, ignore_aliases=false], types=[message], routing='null', preference='null', requestCache=null, scroll=null, maxConcurrentShardRequests=10, batchedReduceSize=512, preFilterShardSize=64, allowPartialSearchResults=true, source={"from":0,"size":0,"query":{"bool":{"filter":[{"query_string":{"query":"Level:ERROR AND Referrer:*find\\-car* AND (NOT messageOrEvent:\"Script error\" OR NOT messageOrEvent:\"Script error.\") AND NOT Browser:InternetExplorer11","fields":[],"type":"best_fields","tie_breaker":0.0,"default_operator":"or","max_determinized_states":10000,"allow_leading_wildcard":true,"enable_position_increments":true,"fuzziness":"AUTO","fuzzy_prefix_length":0,"fuzzy_max_expansions":50,"phrase_slop":0,"escape":false,"auto_generate_synonyms_phrase_query":true,"fuzzy_transpositions":true,"boost":1.0}},{"range":{"timestamp":{"from":"2019-11-14 10:18:26.607","to":"2019-11-14 10:23:26.607","include_lower":true,"include_upper":true,"boost":1.0}}}],"adjust_pure_negative":true,"boost":1.0}},"aggregations":{"timestamp-min":{"min":{"field":"timestamp"}},"timestamp-max":{"max":{"field":"timestamp"}}}}}] lastShard [true]
org.elasticsearch.transport.RemoteTransportException: [graylog4][10.10.3.250:9300][indices:data/read/search[can_match]]
Caused by: org.elasticsearch.ElasticsearchParseException: failed to parse date field [2019-11-14 10:18:26.607] with format [strict_date_optional_time||epoch_millis]
        at org.elasticsearch.common.joda.JodaDateMathParser.parseDateTime(JodaDateMathParser.java:214) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.common.joda.JodaDateMathParser.parse(JodaDateMathParser.java:67) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.index.mapper.DateFieldMapper$DateFieldType.parseToMilliseconds(DateFieldMapper.java:319) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.index.mapper.DateFieldMapper$DateFieldType.isFieldWithinQuery(DateFieldMapper.java:332) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.index.query.RangeQueryBuilder.getRelation(RangeQueryBuilder.java:460) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.index.query.RangeQueryBuilder.doRewrite(RangeQueryBuilder.java:477) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.index.query.AbstractQueryBuilder.rewrite(AbstractQueryBuilder.java:279) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.index.query.BoolQueryBuilder.rewriteClauses(BoolQueryBuilder.java:474) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.index.query.BoolQueryBuilder.doRewrite(BoolQueryBuilder.java:441) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.index.query.AbstractQueryBuilder.rewrite(AbstractQueryBuilder.java:279) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.search.builder.SearchSourceBuilder.rewrite(SearchSourceBuilder.java:921) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.search.builder.SearchSourceBuilder.rewrite(SearchSourceBuilder.java:80) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.index.query.Rewriteable.rewrite(Rewriteable.java:68) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.index.query.Rewriteable.rewrite(Rewriteable.java:51) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.search.internal.ShardSearchLocalRequest$RequestRewritable.rewrite(ShardSearchLocalRequest.java:307) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.search.internal.ShardSearchLocalRequest$RequestRewritable.rewrite(ShardSearchLocalRequest.java:297) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.index.query.Rewriteable.rewrite(Rewriteable.java:68) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.search.SearchService.createSearchContext(SearchService.java:668) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.search.SearchService.canMatch(SearchService.java:1024) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.action.search.SearchTransportService$12.messageReceived(SearchTransportService.java:417) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.action.search.SearchTransportService$12.messageReceived(SearchTransportService.java:414) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler$1.doRun(SecurityServerTransportInterceptor.java:251) ~[?:?]
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler.messageReceived(SecurityServerTransportInterceptor.java:309) ~[?:?]
        at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:66) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.transport.TcpTransport$RequestHandler.doRun(TcpTransport.java:1350) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.common.util.concurrent.EsExecutors$1.execute(EsExecutors.java:135) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.transport.TcpTransport.handleRequest(TcpTransport.java:1308) [elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.transport.TcpTransport.messageReceived(TcpTransport.java:1172) [elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.transport.netty4.Netty4MessageChannelHandler.channelRead(Netty4MessageChannelHandler.java:65) [transport-netty4-client-6.5.4.jar:6.5.4]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:323) [netty-codec-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:310) [netty-codec-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:426) [netty-codec-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:278) [netty-codec-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.handler.logging.LoggingHandler.channelRead(LoggingHandler.java:241) [netty-handler-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1434) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:965) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:644) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:544) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:498) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:458) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:897) [netty-common-4.1.30.Final.jar:4.1.30.Final]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_222]
Caused by: java.lang.IllegalArgumentException: Unrecognized chars at the end of [2019-11-14 10:18:26.607]: [ 10:18:26.607]
        at org.elasticsearch.common.joda.JodaDateMathParser.parseDateTime(JodaDateMathParser.java:210) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.common.joda.JodaDateMathParser.parse(JodaDateMathParser.java:67) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.index.mapper.DateFieldMapper$DateFieldType.parseToMilliseconds(DateFieldMapper.java:319) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.index.mapper.DateFieldMapper$DateFieldType.isFieldWithinQuery(DateFieldMapper.java:332) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.index.query.RangeQueryBuilder.getRelation(RangeQueryBuilder.java:460) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.index.query.RangeQueryBuilder.doRewrite(RangeQueryBuilder.java:477) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.index.query.AbstractQueryBuilder.rewrite(AbstractQueryBuilder.java:279) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.index.query.BoolQueryBuilder.rewriteClauses(BoolQueryBuilder.java:474) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.index.query.BoolQueryBuilder.doRewrite(BoolQueryBuilder.java:441) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.index.query.AbstractQueryBuilder.rewrite(AbstractQueryBuilder.java:279) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.search.builder.SearchSourceBuilder.rewrite(SearchSourceBuilder.java:921) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.search.builder.SearchSourceBuilder.rewrite(SearchSourceBuilder.java:80) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.index.query.Rewriteable.rewrite(Rewriteable.java:68) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.index.query.Rewriteable.rewrite(Rewriteable.java:51) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.search.internal.ShardSearchLocalRequest$RequestRewritable.rewrite(ShardSearchLocalRequest.java:307) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.search.internal.ShardSearchLocalRequest$RequestRewritable.rewrite(ShardSearchLocalRequest.java:297) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.index.query.Rewriteable.rewrite(Rewriteable.java:68) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.search.SearchService.createSearchContext(SearchService.java:668) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.search.SearchService.canMatch(SearchService.java:1024) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.action.search.SearchTransportService$12.messageReceived(SearchTransportService.java:417) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.action.search.SearchTransportService$12.messageReceived(SearchTransportService.java:414) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler$1.doRun(SecurityServerTransportInterceptor.java:251) ~[?:?]
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler.messageReceived(SecurityServerTransportInterceptor.java:309) ~[?:?]
        at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:66) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.transport.TcpTransport$RequestHandler.doRun(TcpTransport.java:1350) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.common.util.concurrent.EsExecutors$1.execute(EsExecutors.java:135) [elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.transport.TcpTransport.handleRequest(TcpTransport.java:1308) ~[elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.transport.TcpTransport.messageReceived(TcpTransport.java:1172) [elasticsearch-6.5.4.jar:6.5.4]
        at org.elasticsearch.transport.netty4.Netty4MessageChannelHandler.channelRead(Netty4MessageChannelHandler.java:65) ~[?:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) ~[?:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) ~[?:?]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) ~[?:?]
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:323) ~[?:?]
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:310) ~[?:?]
        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:426) ~[?:?]
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:278) ~[?:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) ~[?:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) ~[?:?]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) ~[?:?]
        at io.netty.handler.logging.LoggingHandler.channelRead(LoggingHandler.java:241) ~[?:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) ~[?:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) ~[?:?]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) ~[?:?]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1434) ~[?:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) ~[?:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) ~[?:?]
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:965) ~[?:?]
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163) ~[?:?]
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:644) ~[?:?]
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:544) ~[?:?]
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:498) ~[?:?]
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:458) ~[?:?]
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:897) ~[?:?]
        at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_222]
```

**Comment by kmerz** [comment]
I can reproduce the problem but only without a stream.

**Comment by mpfz0r** [comment]
@cw-pratik #7619  is likely the reason for your the problem you're seeing.
It should disappear if you specifically specify one or more streams in the event definition.
Can you confirm this?

**Comment by cw-pratik** [comment]
@mpfz0r 
I checked. You were correct. The errors disappear when I add a stream to the event definition.

**Comment by mpfz0r** [comment]
@cw-pratik Thanks for confirming that. Closing this issue, because it's fixed in 3.2.3 and the upcoming 3.3
